File: frontend/src/lib/review.ts
import { PartialDocumentFields } from "@/lib/chat";

export type RiskSeverity = "high" | "medium" | "low";

export interface ReviewRisk {
  severity: RiskSeverity;
  field: string;
  message: string;
}

export interface DocumentReviewResponse {
  completeness_score: number;
  risks: ReviewRisk[];
  suggestions: string[];
}

export async function reviewDocument(
  documentType: string,
  fields: PartialDocumentFields,
): Promise<DocumentReviewResponse> {
  const res = await fetch("/api/review-document", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    credentials: "include",
    body: JSON.stringify({ document_type: documentType, fields }),
  });
  if (!res.ok) {
    const err = await res.json().catch(() => ({}));
    throw new Error((err as { detail?: string }).detail ?? "Failed to review document");
  }
  return res.json() as Promise<DocumentReviewResponse>;
}

File: backend/schemas/review.py
"""Schemas for the document review endpoint."""
from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, field_validator

from schemas.document import PartialDocumentFields


class ReviewRisk(BaseModel):
    severity: Literal["high", "medium", "low"]
    field: str
    message: str


class AiDocumentReview(BaseModel):
    completeness_score: int
    risks: list[ReviewRisk]
    suggestions: list[str]


class DocumentReviewRequest(BaseModel):
    document_type: str
    fields: PartialDocumentFields

    @field_validator("document_type")
    @classmethod
    def validate_document_type(cls, v: str) -> str:
        from document_registry import DOCUMENT_REGISTRY

        if v not in DOCUMENT_REGISTRY:
            raise ValueError(f"Unsupported document type: {v}")
        return v
